src/UIAutomation/UiaOperationAbstraction/UiaOperationAbstraction.cpp

```diff
@@ -2316,22 +2316,37 @@ namespace UiaOperationAbstraction
 
     void UiaOperationScope::Resolve()
     {
-        HRESULT operationResult = S_OK;
-        ResolveHrInternal(operationResult);
-        THROW_IF_FAILED(operationResult);
+        auto status = winrt::Windows::UI::UIAutomation::Core::AutomationRemoteOperationStatus::Success;
+        HRESULT extendedError = S_OK;
+        ResolveInternal(status, extendedError);
+        switch(status)
+        {
+            case winrt::Windows::UI::UIAutomation::Core::AutomationRemoteOperationStatus::MalformedBytecode:
+            throw MalformedBytecodeException(extendedError);
+            case winrt::Windows::UI::UIAutomation::Core::AutomationRemoteOperationStatus::InstructionLimitExceeded:
+            throw InstructionLimitExceededException(extendedError);
+            case winrt::Windows::UI::UIAutomation::Core::AutomationRemoteOperationStatus::UnhandledException:
+            throw UnhandledRemoteException(extendedError);
+            case winrt::Windows::UI::UIAutomation::Core::AutomationRemoteOperationStatus::ExecutionFailure:
+            throw ExecutionFailureException(extendedError);
+            default:
+            THROW_IF_FAILED(extendedError);
+        }
     }
 
     [[nodiscard]] HRESULT UiaOperationScope::ResolveHr() noexcept try
     {
-        HRESULT operationResult = S_OK;
-        ResolveHrInternal(operationResult);
-        return operationResult;
+        auto status = winrt::Windows::UI::UIAutomation::Core::AutomationRemoteOperationStatus::Success;
+        HRESULT extendedError = S_OK;
+        ResolveInternal(status, extendedError);
+        return extendedError;
     }
     CATCH_RETURN();
   
-    void UiaOperationScope::ResolveHrInternal(HRESULT& operationResult)
+    void UiaOperationScope::ResolveInternal(winrt::Windows::UI::UIAutomation::Core::AutomationRemoteOperationStatus& status, HRESULT& extendedError)

```

Comment:
While we're here could we change this to return an `std::pair<AutomationRemoteOperationState, HRESULT>` instead of using out params?